common/oauthTokenManager.go

```diff
@@ -771,6 +770,29 @@ func (credInfo *OAuthTokenInfo) GetTokenCredential() (azcore.TokenCredential, er
 	}
 }
 
+// MarshalJSON customizes the JSON encoding for AutoLoginType
+func (alt AutoLoginType) MarshalJSON() ([]byte, error) {
```

Comment:
This should ideally be defined alongside the AutoLoginType definition

Reply:
moved MarshalJSON and UnmarshalJSON to environment.go file